### PR TITLE
Trilinos fix epetra deprecations

### LINKED ISSUE #11402: Panzer: ThyraEpetraAdapters does not enable and causes the build to fail
## Bug Report
I noticed my nightly build for MrHyDE (which uses Trilinos on develop as a library) now fails due to a recent change in Panzer, which I suspect is related to #11383. The failure is as follows:
```
In file included from /ascldap/users/MrHyDE/nightly/Trilinos/Trilinos-source/packages/panzer/disc-fe/src/lof/Panzer_BlockedEpetraLinearObjFactory.hpp:57,
                 from /ascldap/users/MrHyDE/nightly/Trilinos/Trilinos-source/packages/panzer/mini-em/src/eqn_sets/MiniEM_AuxiliaryEquationSet_MassMatrix_impl.hpp:25,
                 from /ascldap/users/MrHyDE/nightly/Trilinos/Trilinos-source/packages/panzer/mini-em/src/eqn_sets/MiniEM_AuxiliaryEquationSet_MassMatrix.cpp:4:
/ascldap/users/MrHyDE/nightly/Trilinos/Trilinos-source/packages/panzer/disc-fe/src/lof/Panzer_EpetraLinearObjContainer.hpp:61:10: fatal error: Thyra_EpetraLinearOp.hpp: No such file or directory
   61 | #include "Thyra_EpetraLinearOp.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

### Description
After some additional digging, I found out my build no longer enables `ThyraEpetraAdapters` as Panzer does not require it, but Panzer attempts to build certain Epetra objects that require ThyraEpetraAdapters anyway.

### Steps to Reproduce
Trilinos Develop SHA: 1e393b02bb34862a63983b954e51de8b76fc73d1
Here's the relevant lines from the configure script I was using
```
  -D Trilinos_ENABLE_Amesos:BOOL=ON \
  -D Trilinos_ENABLE_Amesos2:BOOL=ON \
  -D Trilinos_ENABLE_AztecOO:BOOL=ON \
  -D Trilinos_ENABLE_Belos:BOOL=ON \
  -D Trilinos_ENABLE_Compadre:BOOL=ON \
  -D Trilinos_ENABLE_FEI:BOOL=ON \
  -D Trilinos_ENABLE_Ifpack:BOOL=ON \
  -D Trilinos_ENABLE_ML:BOOL=ON \
  -D Trilinos_ENABLE_MueLu:BOOL=ON \
  -D Trilinos_ENABLE_Pamgen:BOOL=ON \
  -D Trilinos_ENABLE_Panzer:BOOL=ON \
    -D Panzer_ENABLE_TESTS:BOOL=OFF \
    -D Panzer_ENABLE_EXAMPLES:BOOL=OFF \
    -D Panzer_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
  -D Trilinos_ENABLE_ROL:BOOL=ON \
  -D Trilinos_ENABLE_SEACAS:BOOL=ON \
    -D SEACASExodus_ENABLE_MPI:BOOL=OFF \
  -D Trilinos_ENABLE_Shards:BOOL=ON \
  -D Trilinos_ENABLE_STK:BOOL=ON \
    -D STK_ENABLE_TESTS:BOOL=OFF \
  -D Trilinos_ENABLE_Stratimikos:BOOL=ON \
  -D Trilinos_ENABLE_Teko:BOOL=ON \
  -D Trilinos_ENABLE_Teuchos:BOOL=ON \
  -D Trilinos_ENABLE_Zoltan:BOOL=ON \
```

Here's my final list of enabled packages:
```
Final set of enabled top-level packages:  Gtest Kokkos Teuchos KokkosKernels RTOp Sacado Epetra Zoltan Shards Triutils Tpetra TrilinosSS Thyra Xpetra AztecOO Galeri Amesos Pamgen Zoltan2Core Ifpack ML Belos Amesos2 S
EACAS Anasazi Ifpack2 Stratimikos FEI Teko Intrepid2 Compadre STK Phalanx NOX MueLu Rythmos ROL Piro Panzer 39

Final set of enabled packages:  Gtest KokkosCore KokkosContainers KokkosAlgorithms Kokkos TeuchosCore TeuchosParser TeuchosParameterList TeuchosComm TeuchosNumerics TeuchosRemainder TeuchosKokkosCompat TeuchosKokkosC
omm Teuchos KokkosKernels RTOp Sacado Epetra Zoltan Shards Triutils TpetraCore Tpetra TrilinosSS ThyraCore ThyraTpetraAdapters Thyra Xpetra AztecOO Galeri Amesos Pamgen Zoltan2Core Ifpack ML Belos Amesos2 SEACASExodu
s SEACASExodus_for SEACASExoIIv2for32 SEACASNemesis SEACASIoss SEACASChaco SEACASAprepro_lib SEACASSupes SEACASSuplib SEACASSuplibC SEACASSuplibCpp SEACASSVDI SEACASPLT SEACASAlgebra SEACASAprepro SEACASBlot SEACASCo
njoin SEACASEjoin SEACASEpu SEACASCpup SEACASExo2mat SEACASExodiff SEACASExomatlab SEACASExotxt SEACASExo_format SEACASEx1ex2v2 SEACASFastq SEACASGjoin SEACASGen3D SEACASGenshell SEACASGrepos SEACASExplore SEACASMapv
arlib SEACASMapvar SEACASMapvar-kd SEACASMat2exo SEACASNas2exo SEACASZellij SEACASNemslice SEACASNemspread SEACASNumbers SEACASSlice SEACASTxtexo SEACASEx2ex1v2 SEACAS Anasazi Ifpack2 Stratimikos FEI Teko Intrepid2 C
ompadre STKUtil STKCoupling STKMath STKSimd STKNGP_TEST STKTopology STKMesh STKIO STKSearch STKTransfer STKTools STKBalance STKUnit_test_utils STKSearchUtil STKUnit_tests STKDoc_tests STKExprEval STKEmend STK Phalanx
 NOX MueLu Rythmos ROL Piro PanzerCore PanzerDofMgr PanzerDiscFE PanzerAdaptersSTK PanzerMiniEM Panzer 120

Final set of non-enabled top-level packages:  TrilinosFrameworkTests TrilinosATDMConfigTests MiniTensor EpetraExt Domi Isorropia Pliris ShyLU_Node Komplex TriKota Intrepid Percept Krino Moertel Zoltan2Sphynx Zoltan2 
ShyLU_DD ShyLU Tempus Stokhos PyTrilinos NewPackage Adelus TrilinosCouplings Pike TrilinosBuildStats TrilinosInstallTests 27

Final set of non-enabled packages:  TrilinosFrameworkTests TrilinosATDMConfigTests KokkosSimd MiniTensor EpetraExt TpetraTSQR Domi ThyraEpetraAdapters ThyraEpetraExtAdapters Isorropia Pliris ShyLU_NodeHTS ShyLU_NodeT
acho ShyLU_NodeBasker ShyLU_NodeFastILU ShyLU_Node SEACASExotec2 Komplex TriKota Intrepid Percept Krino Moertel Zoltan2Sphynx Zoltan2 ShyLU_DDFROSch ShyLU_DDCore ShyLU_DDCommon ShyLU_DD ShyLU Tempus Stokhos PanzerExp
rEval PyTrilinos NewPackage Adelus TrilinosCouplings PikeBlackBox PikeImplicit Pike TrilinosBuildStats TrilinosInstallTests 4
```

I believe this can be fixed by adding something like the following to my configure script
```
  -D Trilinos_ENABLE_Thyra:BOOL=ON \
    -D Thyra_ENABLE_EpetraAdapters:BOOL=ON \
```

However, I'm waiting for the build to finish in order to confirm. The more important thing is that Panzer with Epetra support on develop seems to configure properly and then result in a build failure unless ThyraEpetraAdapters is explicitly enabled by the user.

Edit: Despite the configure complaining `Thyra_ENABLE_EpetraAdapters` was incorrect/unused, it seems that manually enabling Thyra is enough to fix the build.

### PR DESCRIPTION
**Pull request by rppawlo**
@trilinos/teko @trilinos/panzer @trilinos/nox 

## Motivation

Fix issues with deprecated epetra code.

Closes #11402 #11405 #11408

### CODE REVIEW

File: packages/teko/CMakeLists.txt
Comment (cgcgcg) [inline packages/teko/CMakeLists.txt:34]: "AztecOO, Ifpack, Amesos, ML" missing in docstring

**Comment by cgcgcg** [comment]
@rppawlo I think that `Teko_ProbingPreconditionerFactory` https://github.com/trilinos/Trilinos/blob/develop/packages/teko/src/Teko_ProbingPreconditionerFactory.hpp needs to be added here to avoid issues with Isorropia=ON, EpetraAdapters=OFF
https://github.com/trilinos/Trilinos/blob/281650bb58e683daff5c41c90e22bbe3c407afc0/packages/teko/src/CMakeLists.txt#L54-L64

**Review by GrahamBenHarper — COMMENTED** [review]
Thanks so much for doing this @rppawlo. I'll let Christian decide if this is fine to merge or if the changes are necessary.

**Comment by rppawlo** [comment]
@cgcgcg good catch on the documentation and the ProbingPreconditonerFactory! I've made both the changes you requested. Can you review again?

**Review by cgcgcg — APPROVED** [review]
LGTM

**Comment by cgcgcg** [comment]
Thank you, @rppawlo !

**Review by mperego — APPROVED** [review]
Looks good to me. Thanks @rppawlo !

**Pull request merged**
